feature/migration/provider/src/main/kotlin/app/k9mail/feature/migration/provider/SettingsProvider.kt

```diff
@@ -0,0 +1,178 @@
+            ParcelFileDescriptor.AutoCloseOutputStream(writeFileDescriptor).use { outputStream ->
+                settingsExporter.exportPreferences(
+                    outputStream,
+                    includeGlobals = true,
+                    accountUuids,
+                    includePasswords = true,
+                )
+            }
+        }
+
+        return readFileDescriptor
+    }
+
+    override fun insert(uri: Uri, values: ContentValues?): Uri? {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun query(
+        uri: Uri,
+        projection: Array<out String>?,
+        selection: String?,
+        selectionArgs: Array<out String>?,
+        sortOrder: String?,
+    ): Cursor? {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun update(uri: Uri, values: ContentValues?, selection: String?, selectionArgs: Array<out String>?): Int {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    override fun delete(uri: Uri, selection: String?, selectionArgs: Array<out String>?): Int {
+        throw UnsupportedOperationException("not implemented")
+    }
+
+    /**
+     * A trusted caller must appear exactly in our allowlist map - its package map must map to a known signature.
+     * In case of any deviation (multiple signers, certificate rotation), assume that the caller isn't trusted.
+     */
+    // Based on https://searchfox.org/mozilla-esr68/source/mobile/android/services/src/main/java/org/mozilla/gecko/fxa/AuthStateProvider.java
+    @Suppress("ReturnCount")
+    private fun isTrustedCaller(): Boolean {
+        val packageManager = context!!.packageManager
+
+        // Signatures can be easily obtained locally. For an APK in question, unzip it and run:
+        // keytool -printcert -file META-INF/SIGNATURE.RSA
+        // SHA256 certificate fingerprint is what's listed below.
+
+        // We will only service query requests from callers that exactly match our allowlist.
+        // Allowlist is local to this function to avoid exposing it to the world more than necessary.
+        val packageAllowlist = listOf(
+            // K-9 Mail (our signing key)
+            "com.fsck.k9" to "55c8a523b97335f5bf60dfe8a9f3e1dde744516d9357e80a925b7b22e4f55524",
+            // K-9 Mail (F-Droid)
+            "com.fsck.k9" to "c430665e3662253b2078dcda350c2c6ce44d915a3d8a147b63ced619bb9e8576",
+            // Thunderbird for Android (release)
+            "net.thunderbird.android" to "b6524779b3dbbc5ac17a5ac271ddb29dcfbf723578c238e03c3c217811356dd1",
+            // Thunderbird for Android (beta)
+            "net.thunderbird.android.beta" to "056bfafb450249502fd9226228704c2529e1b822da06760d47a85c9557741fbd",
+            // Thunderbird for Android (daily)
+            "net.thunderbird.android.daily" to "c48d74a75c45cd362b0ff2c1e9756f541dee816163e3684a9fd59f6c3ae949b2",
+        )
```

Comment:
This could be moved to another class to manage the allowlist independently from `isTrustedCaller`

For example:

```suggestion
        val packageAllowlist = AllowListManager.getAllowedPackageList()
```

and

```Kotlin
object AllowListManager {
    fun getAllowedPackageList() = listOf(
            // K-9 Mail (our signing key)
            "com.fsck.k9" to "55c8a523b97335f5bf60dfe8a9f3e1dde744516d9357e80a925b7b22e4f55524",
            // K-9 Mail (F-Droid)
            "com.fsck.k9" to "c430665e3662253b2078dcda350c2c6ce44d915a3d8a147b63ced619bb9e8576",
            // Thunderbird for Android (release)
            "net.thunderbird.android" to "b6524779b3dbbc5ac17a5ac271ddb29dcfbf723578c238e03c3c217811356dd1",
            // Thunderbird for Android (beta)
            "net.thunderbird.android.beta" to "056bfafb450249502fd9226228704c2529e1b822da06760d47a85c9557741fbd",
            // Thunderbird for Android (daily)
            "net.thunderbird.android.daily" to "c48d74a75c45cd362b0ff2c1e9756f541dee816163e3684a9fd59f6c3ae949b2",
    )
}
```

Reply:
I was considering that. But I don't think we'll be changing or reusing that list anytime soon, if ever. If we change it, it'll probably be because of #8077. But then we'll most likely have to change the code as well. So it makes sense for the two things to be close together.

We can still extract that list and/or the `isTrustedCaller()` code when there's an actual need for it.